aea/protocols/base.py

```diff
@@ -56,23 +56,42 @@ def __str__(self):
             """Get the string representation."""
             return str(self.value)
 
+    class _SlotsCls:  # pylint: disable=too-few-public-methods
+        __slots__: Tuple[str, ...] = (
+            "performative",
+            "dialogue_reference",
+            "message_id",
+            "target",
+            "content",
+        )
+
+    _performatives: Set[str] = set()
+
     def __init__(self, _body: Optional[Dict] = None, **kwargs):
         """
         Initialize a Message object.
 
         :param body: the dictionary of values to hold.
         :param kwargs: any additional value to add to the body. It will overwrite the body values.
         """
+        self._slots = self._SlotsCls()
+
         self._to: Optional[Address] = None
         self._sender: Optional[Address] = None
-        self.__body: Dict[str, Any] = copy(_body) if _body else {}
-        self.__body.update(kwargs)
+
+        self._update_slots_from_dict(copy(_body) if _body else {})
```

Comment:
Why copy?

Reply:
it was the previous implementation, so decided to stay as is, probably to be sure data passed will not be changed inside the message with side effects